### PR TITLE
Add option to deployableConfig to remove existing resources

### PR DESCRIPTION
**Pull request by diemtvu**
This change allows to write e2e test that needs to modify existing resources. One of example is for [PR 7750](https://github.com/istio/istio/pull/7750), which need to change the default authentication policy and destination rules.

### CODE REVIEW

File: tests/util/kube_utils.go [hunk 1]
```diff
@@ -150,6 +150,18 @@ func KubeApply(namespace, yamlFileName string, kubeconfig string) error {
 	return err
 }
 
+// KubeExport kubectl get yaml content for given resource.
+func KubeExport(namespace, resource, name string, kubeconfig string) (string, error) {
+	var cmd string
+	if namespace == "" {
+		cmd = fmt.Sprintf("kubectl get %s %s -o yaml --kubeconfig=%s --export", resource, name, kubeconfig)
```
Comment (baodongli) [inline tests/util/kube_utils.go]: Consider:
if namespace ==  "" {
    namespace = "default"
}

cmd := .....

File: tests/util/kube_utils.go [hunk 2]
```diff
@@ -150,6 +150,18 @@ func KubeApply(namespace, yamlFileName string, kubeconfig string) error {
 	return err
 }
 
+// KubeExport kubectl get yaml content for given resource.
+func KubeExport(namespace, resource, name string, kubeconfig string) (string, error) {
```
Comment (baodongli) [inline tests/util/kube_utils.go]: nit: KubeGetYaml instead?

File: tests/e2e/tests/pilot/pilot_test.go [hunk 3]
```diff
@@ -152,16 +152,43 @@ func runRetriableTest(t *testing.T, cluster, testName string, retries int, f fun
...
 // deployableConfig is a collection of configs that are applied/deleted as a single unit.
 type deployableConfig struct {
-	Namespace  string
-	YamlFiles  []string
+	Namespace string
+	YamlFiles []string
+	// List of resources must be removed during deployableConfig setup, and restored
+	// during teardown. Typically, these are resources added by default installation and need to
+	// be modified for test. These resources are in the same namespace defined above.
+	Removes    []resource
```
Comment (baodongli) [inline tests/e2e/tests/pilot/pilot_test.go]: Are users of this structure required to populate the 'Removes' list? Maybe you want to point it out in the comments.

**Comment by diemtvu** [comment]
@baodongli, can you take another look? Thanks

**Comment by baodongli** [comment]
The PR description indicates that PR 7750 would need the feature added by this PR. Checking 7750, I am not seeing the change related to this PR. Is that particular change going to be added after this PR?

**Comment by wattli** [comment]
I will wait for this PR to check in and rebase #7750

**Pull request merged**